.github/workflows/update.yml

```diff
@@ -25,35 +25,96 @@ jobs:
             branch: "nixos-24.05"
           - selectedBranch: "nixos-24.05"
             branch: main
+
     name: Update the flake inputs and generate options
     runs-on: ubuntu-latest
     timeout-minutes: 40
+
+    permissions:
+      contents: write
+      pull-requests: write
+
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          token: ${{ secrets.GH_TOKEN_FOR_UPDATES }}
-          ref: "${{ matrix.branch }}"
+          ssh-key: ${{ secrets.UPDATE_CI_KEY }}
```

Comment:
Repo secret `UPDATE_CI_KEY` needs to contain the SSH private key.

The corresponding public key should be added as a "Deploy Key" with **write** access.

See [instructions here](https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#push-using-ssh-deploy-keys).

Reply:
Can you rename it to `SSH_DEPLOY_KEY` ?